### PR TITLE
[Inductor] aten.dist decomposition

### LINKED ISSUE #105557: aten.dist

### PR DESCRIPTION
**Pull request by yanboliang**
Fixes #105557

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @ngimel @yf225 @chenyang78 @kadeng @muchulee8 @aakhundov

### CODE REVIEW

**Review by desertfire — APPROVED** [review]

File: test/inductor/test_torchinductor.py [hunk 1]
```diff
@@ -976,6 +976,12 @@ def fn(a):
 
         self.common(fn, (torch.randint(4, (4,)),))
 
+    def test_dist(self):
+        def fn(a, b):
+            return torch.dist(a, b)
```
Comment (Chillee) [inline test/inductor/test_torchinductor.py:981]: It would be good to add some other cases as well, such as different `p` or different dtypes.

**Review by Chillee — APPROVED** [review]

File: torch/_decomp/decompositions.py [hunk 2]
```diff
@@ -630,6 +630,11 @@ def soft_margin_loss_backward(
     return grad_input
 
 
+@register_decomposition(aten.dist)
+def dist(input: Tensor, other: Tensor, p: float = 2):
```
Comment (lezcano) [inline torch/_decomp/decompositions.py:634]: This would be better suited for a ref, but fine

Comment (Chillee) [reply torch/_decomp/decompositions.py:634]: This was done as part of a brief hackathon, so we were looking for the smallest tasks possible :P

**Pull request closed without merging**